include/aws/common/windows/rw_lock.inl

```diff
@@ -0,0 +1,82 @@
+#ifndef AWS_COMMON_WINDOWS_RW_LOCK_INL
```

Comment:
I don't support of worrying about inlines here. At the point you're grabbing a lock, do you actually give a damn about a function call overhead?

Reply:
I was just doing what @graebm suggested. You guys fight it out.